governance/src/components/FAB/StakingFab.tsx

```diff
@@ -59,10 +63,32 @@ const FABOffsetContainer = styled.div`
 const StakingFAB = () => {
   const { active } = useWeb3React();
   const [, setStakingModal] = useGovernanceGlobalState("stakingModal");
+  const [, setUnstakingModal] = useGovernanceGlobalState("unstakingModal");
 
   const rbnAllowance = useTokenAllowance("rbn", VotingEscrowAddress);
-  const { data: rbnAccount, loading: rbnAccountLoading } = useRBNTokenAccount();
-  const loadingText = useTextAnimation(rbnAccountLoading);
+
+  const { data: rbnTokenAccount, loading: rbnTokenAccountLoading } =
+    useRBNTokenAccount();
+  const { balance: veRBNBalance, loading: votingPowerLoading } =
+    useAssetBalance("veRBN");
+  const loading = rbnTokenAccountLoading || votingPowerLoading;
+  const loadingText = useTextAnimation(loading);
+
+  const stakeMode = useMemo(() => {
+    if (rbnAllowance?.isZero()) {
```

Comment:
Can do `const rbnAllowance = useTokenAllowance("rbn", VotingEscrowAddress) || BigNumber.from(0);` to avoid the `?`